packages/api/src/index.ts

```diff
@@ -57,6 +67,34 @@ const getApp = async () => {
 		}),
 	]);
 
+	apiRouter.get('/signedUrl', [
+		checkAuth,
+		asyncHandler(async (req, res) => {
+			const key = uuidv4();
+			const queryParams = SignedUrlQueryParams.safeParse(req.query);
+			if (!queryParams.success) {
+				res.status(422).send('missing query parameters');
+				return;
+			}
+
+			const presignedS3Url = await getSignedUrl(
+				s3Client,
+				new PutObjectCommand({
```

Comment:
This is a nitpick but as a pattern I think it's good to keep code using the aws sdk within the relevant file (in this case `s3.ts`. Here you could pass Bucket/Key/Metadata into getSignedUrl and generate the PutObjectCommand there.

The only actual reason I can think for doing this is that if in some alternative universe where we write some tests, it will be easier to test the `getSignedUrl` function rather than having to test the entire `/signedUrl` controller

Reply:
good idea - I've moved to s3.ts